### PR TITLE
✏️ Invert order of `typing` <-> `typing_extensions`

### PR DESCRIPTION
**Pull request by Kludex**
The description of the error message is not accurate.

Selected Reviewer: @adriangb

### CODE REVIEW

**Comment by Kludex** [comment]
please review

File: docs/usage/errors.md [hunk 1]
```diff
@@ -290,8 +290,7 @@ except PydanticUserError as exc_info:
 
 ## `TypedDict` version {#typed-dict-version}
 
-This error is raised when you use `typing_extensions.TypedDict`
-instead of `typing.TypedDict` on Python < 3.11.
+This error is raised when you use `typing.TypedDict` instead of `typing_extensions.TypedDict` on Python < 3.12.
```
Comment (hramezani) [inline docs/usage/errors.md:293]: Shouldn't it be `Python < 3.11`?

There is a [note in docstring of the method](https://github.com/pydantic/pydantic/blob/35e9fd00cc436e4898dd41a065479618753eead6/pydantic/_internal/_generate_schema.py#L784-L785) that says  `we only support typing.TypedDict on Python >= 3.11`.

But in the [error message](https://github.com/pydantic/pydantic/blob/35e9fd00cc436e4898dd41a065479618753eead6/pydantic/_internal/_generate_schema.py#L797) of the same function, we have 3.12

I think we need to change the docstring or error message

Comment (Kludex) [reply docs/usage/errors.md:293]: The docstring is wrong.

Comment (hramezani) [reply docs/usage/errors.md:293]: Will you fix it in this commit?

Comment (Kludex) [reply docs/usage/errors.md:293]: Yes

Comment (Kludex) [reply docs/usage/errors.md]: Ok, it looks like it's supposed to be < 3.11 instead of < 3.12.

**Comment by Kludex** [comment]
please review

File: docs/usage/errors.md (hunk 1 above)
Comment (adriangb) [reply docs/usage/errors.md]: It needs to be 3.12 for TypedDic

**Comment by adriangb** [comment]
Superseded by #6370

**Pull request closed without merging**